MantidPlot/src/ProjectRecoveryGUIs/ProjectRecoveryModel.cpp

```diff
@@ -0,0 +1,219 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ProjectRecoveryModel.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidKernel/ConfigService.h"
+#include "ProjectRecovery.h"
+#include "ProjectRecoveryPresenter.h"
+#include "RecoveryThread.h"
+
+#include <Poco/File.h>
+#include <Poco/Path.h>
+#include <QApplication>
+#include <QThread>
+#include <fstream>
+#include <memory>
+
+namespace {
+std::string findNumberOfWorkspacesInDirectory(Poco::Path path) {
+  std::vector<std::string> files;
+  Poco::File(path).list(files);
+  // Number of workspaces is equal to the number of files in the path directory
+  // -1 of that value.
+  return std::to_string(files.size() - 1);
+}
+} // namespace
+
+ProjectRecoveryModel::ProjectRecoveryModel(
+    MantidQt::ProjectRecovery *projectRecovery,
+    ProjectRecoveryPresenter *presenter)
+    : m_projRec(projectRecovery), m_presenter(presenter) {
+  fillFirstRow();
+  // Default to failed run
+  m_failedRun = true;
+  m_recoveryRunning = false;
+}
+
+std::vector<std::string> ProjectRecoveryModel::getRow(int i) {
```

Comment:
Would it make sense for this to return a const reference instead, i.e. `const std::vector<std::string> &`?

Reply:
:+1: